neutromeratio/ani.py

```diff
@@ -123,14 +88,82 @@ def calculate_force(self, x:simtk.unit.quantity.Quantity) -> simtk.unit.quantity
         return F * (unit.kilojoule_per_mole / unit.nanometer), energy_in_kJ_mol.item() * unit.kilojoule_per_mole
 
     
-    def calculate_energy(self, x:simtk.unit.quantity.Quantity) -> simtk.unit.quantity.Quantity:
+    def _calculate_energy(self, coordinates:torch.tensor, lambda_value:float = 0.0)->torch.tensor:
+        """
+        Helpter function to return energies as tensor.
+        Given a coordinate set the energy is calculated.
+        
+        Parameters
+        ----------
+        coordinates : torch.tensor 
+            coordinates in nanometer without units attached
+        lambda_value : float
+            between 0.0 and 1.0
+        Returns
+        -------
+        energy_in_kJ_mol : torch.tensor
+            return the energy with restraints added
+        """
+
+        
+        assert(float(lambda_value) <= 1.0 and float(lambda_value) >= 0.0)
+
+        # coordinates in nm!
+        
+        if self.use_pure_ani1ccx:
+            _, energy_in_hartree = self.model((self.species, coordinates * nm_to_angstroms))
+        else:
+            _, energy_in_hartree = self.model((self.species, coordinates * nm_to_angstroms, lambda_value))
+        
+        # convert energy from hartrees to kJ/mol
+        energy_in_kJ_mol = energy_in_hartree * hartree_to_kJ_mol
+
+        bias_flat_bottom = 0.0
+        bias_harmonic = 0.0
+        bias = 0.0
+
+        if self.flat_bottom_restraint:
```

Comment:
instead of having if/else logic here to decide for each restraint whether to compute `restraint.flat_bottom_position_restraint` or `restraint.harmonic_position_restraint`, can we have `for restraint in self.list_of_restraints: e = restraint.compute_energy(x)`?